Dynatrace/dynatrace-operator
# Improve performance of namespace secret generation

# Description

Handling secrets for multiple namespaces is currently done by checking each secret one by one. 

This leads to performance problems for the operator, lot's of requests to the kubernetes API and log spamming.

This PR addresses the problem by fetching all relevant namespace secrets at once and only updates/creates outdated ones.
Fixes #1251 

## How can this be tested?

- Deploy the operator as usual and a applicationOnly monitoring DynaKube, which monitores multiple namespaces
- In the logs one should see that reconciliation is not slowed down by secret updates

## Checklist
